### DISCUSSION TITLE
Reducing CSS redundancy

### DISCUSSION THREAD

**Discussion by jmgirard** [PittMethods/appliedml#17] [Ideas]
Instead of generating a copy of `xaringan-themer.css` and `style.css` in every folder, let's just create them once in the root folder and refer to them relatively. You can see how I did this in the new version of `day_1a.Rmd`. But in short:

1) Replace `css: [xaringan-themer.css, style.css]` in the YAML header with `css: [../xaringan-themer.css, ../styles.css]`
2) Remove the following code from each Rmd file:
```
style_mono_light(
  base_color = "#23395b",
  footnote_font_size = "20px",
  footnote_color = "gray",
  text_slide_number_font_size = "18px"
)
```
3) Move any custom CSS from the first slide to the dedicated `styles.css` file in root. So remove this:
```
<style type="text/css">
.onecol {
    font-size: 26px;
}
.twocol {
  font-size: 24px;
}
.remark-code {
  font-size: 24px;
  border: 1px solid grey;
}
a {
  background-color: lightblue;
}
.remark-inline-code {
  background-color: white;
}
</style>
```
4) Add `css_file = "../xaringan-themer.css"` to each call of `theme_xaringan()` when using ggplot2

**Comment by jmgirard** [comment]
Similarly, we can remove the `macro.js` file entirely and accomplish the same thing via CSS by:

1. Removing `beforeInit: "macro.js"` from the header
2. Replacing `![:emphasize](text to highlight)` with `.imp[text to highlight]`.

**Comment by jmgirard** [comment]
@ShirleyBWang I took care of this for all existing .Rmd files, but wanted to give you a heads up for future files.

**Reply by ShirleyBWang** [reply]
Awesome - thanks so much for doing that and I'll use this structure for future files!
